pageserver/src/tenant/timeline.rs

```diff
@@ -3217,62 +2911,36 @@ impl Timeline {
             .await
             .context("fsync of timeline dir")?;
 
-        let mut layer_paths_to_upload = HashMap::with_capacity(image_layers.len());
-
         let mut guard = self.layers.write().await;
-        let timeline_path = self.conf.timeline_path(&self.tenant_id, &self.timeline_id);
 
         for l in &image_layers {
-            let path = l.filename();
-            let metadata = timeline_path
-                .join(path.file_name())
-                .metadata()
-                .with_context(|| format!("reading metadata of layer file {}", path.file_name()))?;
-
-            layer_paths_to_upload.insert(path, LayerFileMetadata::new(metadata.len()));
-
+            // FIXME: these should be in guard
             self.metrics
                 .resident_physical_size_gauge
-                .add(metadata.len());
-            let l = Arc::new(l);
-            l.access_stats().record_residence_event(
-                &guard,
-                LayerResidenceStatus::Resident,
-                LayerResidenceEventReason::LayerCreate,
-            );
+                .add(l.layer_desc().file_size);
```

Comment:
Or: not the guard. If I add that better LayerE::new for existing files, metrics could be fired from there, because eviction already is.

Reply:
These are now in LayerFileManager.